domain/removal/state/state.go

```diff
@@ -29,6 +31,62 @@ func NewState(factory database.TxnRunnerFactory, logger logger.Logger) *State {
 	}
 }
 
+// GetAllJobs returns all scheduled removal jobs.
+func (st *State) GetAllJobs(ctx context.Context) ([]removal.Job, error) {
+	db, err := st.DB()
+	if err != nil {
+		return nil, errors.Capture(err)
+	}
+
+	stmt, err := st.Prepare("SELECT &removalJob.* FROM removal", removalJob{})
+	if err != nil {
+		return nil, errors.Errorf("preparing select jobs query: %w", err)
+	}
+
+	var dbJobs []removalJob
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err = tx.Query(ctx, stmt).GetAll(&dbJobs)
+		if err != nil && !errors.Is(err, sqlair.ErrNoRows) {
+			return errors.Errorf("running select jobs query: %w", err)
+		}
+		return nil
+	})
+
+	if err != nil {
+		return nil, errors.Capture(err)
+	}
+
+	if len(dbJobs) == 0 {
+		return nil, nil
+	}
+
+	jobs := make([]removal.Job, len(dbJobs))
+	for i, job := range dbJobs {
+		var arg map[string]any
+		if job.Arg.Valid && job.Arg.String != "" {
+			if err := json.Unmarshal([]byte(job.Arg.String), &arg); err != nil {
+				return nil, errors.Errorf("decoding job arg: %w", err)
+			}
+		}
+
+		jUUID := removal.UUID(job.UUID)
+		if err := jUUID.Validate(); err != nil {
```

Comment:
Only other place I could find that validates UUIDs coming from reads of state is `uuidForName` in the access domain.